plutus-ir/src/Language/PlutusIR/Compiler/Term.hs

```diff
@@ -71,9 +72,9 @@ compileSingleBinding r body b =  case b of
         Rec -> compileRecTerms body [PLC.Def d rhs]
         NonRec -> local (TermBinding (varDeclNameString d)) $ do
             def <- PLC.Def d <$> compileTerm rhs
-            PLC.mkTermLet <$> ask <*> pure def <*> pure body
+            asks PLC.mkTermLet <*> pure def <*> pure body
```

Comment:
:-1: This breaks the regularity which makes it harder to read. Not sure why this isn't in the hlint ignore file, I'll add it.